src/main/kotlin/com/exactpro/th2/codec/configuration/ApplicationContext.kt

```diff
@@ -231,13 +231,18 @@ class ApplicationContext(
             }
         }
 
-        inline fun <reified T> load(): T {
-            val instances = ServiceLoader.load(T::class.java).toList()
+        private inline fun <reified T : Any> load(codecClassName: String?): T {
```

Comment:
```suggestion
        private inline fun <reified T : Any> load(codecClassName: String): T {
```
All sailfish codecs must have codec class name

Reply:
This is nullable because if we have only one codec implementation this one will be loaded. What I can do here if the codecClassName is set then I check that the loaded impl matches the provided name.

The original change removed the usage of that variable completely. So, I made the decision accordingly and do not force that parameter (as it is nullable)